aiida/backends/sqlalchemy/migrations/versions/041a79fc615f_dblog_cleaning.py

```diff
@@ -0,0 +1,182 @@
+    if lwf_no == 0 and other_no == 0:
+        return
+
+    if not settings.TESTING_MODE:
+        click.echo('We found {} log records that correspond to legacy workflows and {} log records to correspond '
+                   'to an unknown entity.'.format(lwf_no, other_no))
+        click.echo(
+            'These records will be removed from the database and exported to JSON files to the current directory).')
+        proceed = click.confirm('Would you like to proceed?', default=True)
+        if not proceed:
+            return
+
+    delete_on_close = False
+    if settings.TESTING_MODE:
+        delete_on_close = True
+
+    # Exporting the legacy workflow log records
+    if lwf_no != 0:
+        # Get the records and write them to file
+        with NamedTemporaryFile(
+                prefix='legagy_wf_logs-', suffix='.log', dir='.', delete=delete_on_close, mode='w+') as handle:
+            # Export the log records
+            handle.write(get_serialized_legacy_workflow_logs(connection))
+        # Now delete the records
+        # connection.execute(
+        connection.execute('DELETE FROM db_dblog WHERE (db_dblog.objname LIKE \'aiida.workflows.user.\' || \'%%\')')
+
+    # Exporting unknown log records
+    if other_no != 0:
+        # Get the records and write them to file
+        with NamedTemporaryFile(
+                prefix='unknown_entity_logs-', suffix='.log', dir='.', delete=delete_on_close, mode='w+') as handle:
+            # Export the log records
+            handle.write(get_serialized_unknown_entity_logs(connection))
+        # Now delete the records
+        # connection.execute(
+        connection.execute(
+            'DELETE FROM db_dblog WHERE (db_dblog.objname NOT LIKE \'node.\' || \'%%\') AND (db_dblog.objname '
+            'NOT LIKE \'aiida.workflows.user.\' || \'%%\')')
+
+
+def upgrade():
+    """
+    Changing the log table columns to use uuid to reference remote objects and log entries.
+    Upgrade function.
+    """
+    connection = op.get_bind()
+
+    # Clean data
+    export_and_clean_workflow_logs(connection)
+
+    # Create the dbnode_id column and add the necessary index
+    op.add_column('db_dblog', sa.Column('dbnode_id', sa.INTEGER(), autoincrement=False, nullable=True))
+    # Transfer data to dbnode_id from objpk
+    connection.execute('UPDATE db_dblog SET dbnode_id=objpk')
+
+    op.create_foreign_key(
+        None,
+        "db_dblog",
+        "db_dbnode", ['dbnode_id'], ['id'],
+        ondelete=u'CASCADE',
+        initially=u'DEFERRED',
+        deferrable=True)
+
+    # Update the dbnode_id column to not nullable
+    op.alter_column('db_dblog', 'dbnode_id', nullable=False)
+
+    # Remove the objpk column
+    op.drop_column('db_dblog', 'objpk')
+
+    # Remove the objname column
+    op.drop_column('db_dblog', 'objname')
+
+    # Remove objpk and objname from metadata dictionary
+    connection.execute('UPDATE db_dblog SET metadata = metadata - \'objpk\' - \'objname\' ')
```

Comment:
use double quotes to simplify SQL also here

Reply:
Done